src/ripple/app/data/SociDB.cpp

```diff
@@ -51,24 +54,21 @@ getSociInit (BasicConfig const& config,
              std::string const& dbName)
 {
     auto const& section = config.section ("sqdb");
-    std::string const backendName(get(section, "backend", "sqlite"));
+    auto const backendName = get(section, "backend", "sqlite");
 
-    if (backendName == "sqlite")
-    {
-        std::string const path = config.legacy ("database_path");
-        std::string const ext =
-            (dbName == "validators" || dbName == "peerfinder") ? ".sqlite"
-                                                               : ".db";
-        return detail::getSociSqliteInit(dbName, path, ext);
-    }
-    else
-    {
+    if (backendName != "sqlite")
```

Comment:
New backends will be added as `else if (backendName == "postgresql") ...`. For that reason I prefer the original version of this code, even if your version is nicer when we only have one backend.